prusti-encoder/src/encoders/util.rs

```diff
@@ -0,0 +1,31 @@
+use vir::VirCtxt;
+use prusti_rustc_interface::{
+    middle::{mir, ty},
+    span::def_id::DefId,
+};
+
+/// Returns input and output types of a function
+pub fn get_func_sig<'tcx>(vcx: &VirCtxt<'tcx>, def_id: DefId) -> (Vec<ty::Ty<'tcx>>, ty::Ty<'tcx>) {
```

Comment:
Why this function? Why do something different with local bodies? And why is `fn_sig` generally not sufficient? I am quite hesitant to have something like this in the codebase, because (apart from another `util` file) it suggests that the type of a method is fully determined by the `DefId`. However, this is only the case in this generic encoding, but is not true for example when monomorphisation is used.

Reply:
I copied the logic from here: https://github.com/Aurel300/prusti-dev/blob/7aaa380c654bd17dada922f189a522f422c0604d/prusti-encoder/src/encoders/local_def.rs#L75-L105

I suppose it might be worth adding a comment there to explain what it's doing? I am not sure why there needs to be a case split there.

Agree with your points, I'll:
- remove this file
- change occurrences of this function to use `fn_sig`

I don't think it should cause any issues (I can't think of any reason why `fn_sig` wouldn't work)